### ISSUE TITLE
[ally]: Combobox Default & with Layer - VO doesn't announce keyboard input

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#14707]
## Environment

> Operating system

> Browser

> Automated testing tool and ruleset

> Assistive technology used to verify

## Detailed description

> What version of the Carbon Design System are you using?

> What did you expect to happen?
If the VO can provide the feedback to the user about the availability of the keyboard and instantaneously announce. the modification that happens upon providing the input, it will be helpful

When the screen reader doesn’t identify the page elements, it impacts the experience of users who are blind or have vision impairment. They can miss important information or fail to perform an action on the web page. 

NOTE: The same issue is observed even in the aria combobox example that is given below.
https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-autocomplete-both/

> What happened instead?
In voice over on iPhone, after double tapping the "edit combo" the keyboard is displayed visually. However, the same is not announced by screen reader. Hence the user may not realize that the input can also be typed.

> What WCAG 2.1 checkpoint does the issue violate?

## Steps to reproduce the issue
Default: https://react.carbondesignsystem.com/?path=/story/components-combobox--default
With layer: https://react.carbondesignsystem.com/?path=/story/components-combobox--with-layer

## Additional information

- Screenshots or code
- Notes
